src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -665,31 +663,20 @@ sendMoney sendActions passphrase moneySource dstDistr = do
 getFullWalletHistory :: WalletWebMode m => CId Wal -> m ([CTx], Word)
 getFullWalletHistory cWalId = do
     addrs <- mapM decodeCIdOrFail =<< getWalletAddrs Ever cWalId
-    cHistory <- do
-        (mInit, cachedTxs) <- transCache <$> getHistoryCache cWalId
-
-        TxHistoryAnswer {..} <- untag @WalletSscType getTxHistory addrs mInit
-
-        -- Add allowed portion of result to cache
-        let fullHistory = taHistory <> cachedTxs
-            lenHistory = length taHistory
-            cached = drop (lenHistory - taCachedNum) taHistory
-        unless (null cached) $
-            updateHistoryCache
-                cWalId
-                taLastCachedHash
-                taCachedUtxo
-                (cached <> cachedTxs)
-
-        ctxs <- forM fullHistory $ addHistoryTx cWalId False
-        pure $ concatMap toList ctxs
+
+    blockHistory <- getHistoryCache cWalId >>= \case
+        Just hist -> pure $ DL.fromList hist
+        Nothing -> do
```

Comment:
does this code make a lot of sense?
I don't understand when it can be called. we synchronise our cache on an import of wallet so it's strange if it is called at all. looks like a piece of old logic.
maybe make sense logging some message here to see that something strange happened